scripts/irods/test/test_iadmin.py

```diff
@@ -1685,6 +1685,43 @@ def test_iadmin_mkgroup_returns_correct_error_on_duplicate_group__issue_7599(sel
         finally:
             self.admin.run_icommand(['iadmin', 'rmgroup', test_group_name])
 
+    def test_iadmin_exits_nonzero_on_error_cases__issue_7734(self):
+        # Nonexistent command
+        out, err, ec = self.admin.run_icommand(['iadmin', 'nonexistentcommand']);
```

Comment:
I don't think `.run_icommand()` prints the executed command to the junit xml output file.

By all means, please take a minute or two to confirm that. If it turns out that `.run_icommand()` doesn't include the executed command in the junit xml output, replace it with `.assert_icommand()`. We know that prints the commands to the file.

**Keep in mind that `ec` becomes the first return value for `.assert_icommand()`.**

To keep `.assert_icommand()` from failing the test, you can give it things to check that are guaranteed to exist in the output. For example:
```py
# For situations where you expect output on stdout, but don't care what it is.
ec, _, _ = self.admin.assert_icommand(cmd, 'STDOUT')

# For situations where you expect output on stderr, but don't care what it is.
ec, _, _ = self.admin.assert_icommand(cmd, 'STDERR')

# For situations where you expect output on stdout and need to pass it something
# that will appear in that output. Same thing for stderr.
ec, _, _ = self.admin.assert_icommand(cmd, 'STDOUT', [' '])
```

Check out some of the other tests for examples. Should be easy to spot them in newer tests.